posts/2021-03-03-jakarta-ee-9-compatibility.adoc

```diff
@@ -0,0 +1,26 @@
+Jared Anderson <https://github.com/jhanders34>
+:imagesdir: /
+:url-prefix: https://openliberty.io
+:url-about: /
+//Blank line here is necessary before starting the body of the post.
+
+Late last year, the Jakarta EE community released the next step in the evolution of the Jakarta EE specification.  link:https://jakarta.ee/specifications/platform/9/jakarta-platform-spec-9.html[Version 9.0 of the Jakarta EE specification] focuses on renaming packages and namespaces by changing to jakarta prefixed packages names in the API and to the link:https://jakarta.ee/xml/ns/jakartaee/[jakarta.ee namespace] in the deployment descriptor files.  Additionally, the specification makes optional or removes older functions that have been deprecated or stabilized for many years.
+
+The Open Liberty development team has been working over the last year to implement the Jakarta EE 9 specification as it was being finalized.  In an effort to not have to duplicate the same code with references to different API package names, the team developed an link:https://projects.eclipse.org/projects/technology.transformer[Open Source byte code transformation technology].  This technology allows for the same code to be compiled to be Jakarta EE 8 compatible and then be transformed to work with Jakarta EE 9 by manipulating the byte code to convert class and property references from being prefixed with javax to jakarta.
+
+As of the link:{url-prefix}/blog/2021/01/26/ee9-messaging-security-21002-beta.html[21.0.0.2-beta release], Open Liberty is the first vendor product added to the link:https://jakarta.ee/compatibility/#tab-9[Jakarta EE website] to be Jakarta EE Web Profile 9.0 compatible.  With the recent link:{url-prefix}/blog/2021/02/19/mpcontextpropagation-requesttiming-21003-beta.html[21.0.0.3-beta release], Open Liberty is the first vendor product to be added to the Jakarta EE Platform 9.0 compatibility list.  These releases continue to demonstrate Open Liberty's leadership in the Jakarta EE technologies.  With the upcoming link:https://jakarta.ee/specifications/platform/9.1/[Jakarta EE 9.1 release] that adds Java 11 support to the TCK, I anticipate that Open Liberty will be added to the compatibility list for the 9.1 release when it is available since link:{url-prefix}/blog/2019/02/06/java-11.html[Open Liberty has supported Java 11] for the last 2 years.
+
+Even with reaching these milestones, the journey to support Jakarta EE 9 in the Open Liberty product will continue until it is available in a non-beta form.  Many Liberty value-add features need to be enabled to run with Jakarta EE 9 and still continue to work with the previous EE specifications.  Over the next months, you will see additional enablement of these features in the beta releases until all are ready.  With Open Liberty's design to be able to enable features in the server.xml, you will be able to run your Java EE 7, Java / Jakarta EE 8 and Jakarta EE 9 applications with the same product install.
```

Comment:
_With Open Liberty's design to be able to enable features in the server.xml_

Is this the same thing as zero migration? If so, recommend:

_Thanks to the Open Liberty [zero migration architecture](https://www.openliberty.io/docs/latest/zero-migration-architecture.html#_open_liberty_features), you will be able..._

or

_Thanks to Open Liberty [pluggable features](https://www.openliberty.io/docs/latest/zero-migration-architecture.html#_open_liberty_features), you will be able..._

Reply:
I combined the two.